### ISSUE TITLE
Problems using a symbol defined in a WITH block in SDL.

### ISSUE THREAD

**Issue by vpetrovykh** [edgedb/edgedb#2320]
- EdgeDB Version: 1.0.0-beta.1
- OS Version: Linux

Steps to Reproduce:

The following schema fails to create a migration:
```
module default {
    type Foo {
        property val -> str;
        property comp := count((
            WITH x := .val
            SELECT y := Bar FILTER x = y.val
        ))
    }

    type Bar {
        property val -> str;
    }
};
```
The error message is:
```
ERROR: InvalidReferenceError: schema item 'std::x' does not exist
edgedb error: cannot proceed until .esdl files are fixed
```

It appears that validation during tracing is at fault.

**Comment by elprans** [comment]
Seems to be a duplicate of #1452
